src/mmw/apps/user/views.py

```diff
@@ -93,10 +93,36 @@ def itsi_auth(request):
     else:
         # User did not authenticate. Save their ITSI ID and send to /register
         request.session['itsi_id'] = itsi_user['id']
-        return redirect('/register')  # TODO Make this Backbone route
+        return redirect('/register/{username}/{first_name}/{last_name}'.format(
+            **itsi_user['extra']
+        ))
 
 
+@decorators.api_view(['POST'])
+@decorators.permission_classes((AllowAny, ))
 def itsi_register(request):
+    # Validate request
+    errors = []
+    if 'itsi_id' not in request.session:
+        errors.append("There was an error in authenticating you with ITSI")
+
+    if 'username' not in request.POST or not request.POST.get('username'):
```

Comment:
YES thank you that's what I was looking for. This stuff is impossible to phrase into a Google query, but easily caught by those more experienced with the language's idioms.